### PR TITLE
43mint2: An omnibus merge

### PR DESCRIPTION
**Pull request by faddat**
So, SDK 0.43.0 adds too much good stuff.  We needed it. 

we also needed the custom minter, so, here they are, combined.  There may be dragons here, but this compiles and can run a single node testnet.  

We can do this, or...........

1) rescaffold again (tho current starport doesn't work for me)
2) merge module into that.  Not sure if sp is on 0.43.0, however.  

master branch sp isn't working correclty for me right now, so this is that.  

Plz do review.  

Thanks.  

**SUMMARY**

* go 1.17
* cosmos-sdk 0.43.0
* encompasses #36 
* preserves spm by replacing it with my faddat/spm, which is just spm, upgraded to 0.43.0

THERE MAY BE DRAGONS BUT I FEEL GOOD ABOUT IT OVERALL :)

### CODE REVIEW

File: x/mint/simulation/params.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	"github.com/cosmos/cosmos-sdk/x/simulation"
+
+	"github.com/CosmosContracts/juno/x/mint/types"
+	simtypes "github.com/cosmos/cosmos-sdk/types/simulation"
+)
+
+const (
+	keyInflationRateChange = "InflationRateChange"
+	keyInflationMax        = "InflationMax"
+	keyInflationMin        = "InflationMin"
```
Comment (giansalex) [inline x/mint/simulation/params.go:18]: keys was deleted

File: x/mint/spec/04_params.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+|---------------------|-----------------|------------------------|
+<<<<<<< HEAD
+| MintDenom           | string          | "uatom"                |
+| InflationRateChange | string (dec)    | "0.130000000000000000" |
+| InflationMax        | string (dec)    | "0.200000000000000000" |
+| InflationMin        | string (dec)    | "0.070000000000000000" |
+| GoalBonded          | string (dec)    | "0.670000000000000000" |
+=======
+| MintDenom           | string          | "ujuno"                |
+>>>>>>> disperze/mint-module
```
Comment (giansalex) [inline x/mint/spec/04_params.md:19]: no clean merge

**Comment by giansalex** [comment]
I don't see the complete changes from #36 🙄 
We need to organize this, can we create pull requests by issue?
- cosmos-sdk 0.43
- disable/remove wasm
- mint-module

**Comment by faddat** [comment]
@giansalex I apologize.  This came out totally wrong in several ways because I worked on it while too tired.  We should merge yours, doing mine a second time, while not butchering your work will be easier than doing it the first time and butchering it.

**Pull request closed without merging**